src/helm/common/file_caches/file_cache.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+
+class FileCache(ABC):
+    """
+    Cache to store files.
+
+    Attributes
+    ----------
+    location: str
+        Location of the files (e.g., local path to a directory or a URL)
+    file_extension: str
+        The types of file we're storing in the cache (e.g., png, jpg, mp4, etc.).
+    binary_mode: bool
+        Whether to write in binary mode (default is True).
+    """
+
+    def __init__(self, location: str, file_extension: str, binary_mode: bool = True):
```

Comment:
None of these parameters should be in the abstract base class:

- location: should only be used by the concrete class
- file_extension: this shouldn't be used at all (unless you're using it for MIME type inference or something)
- binary_mode: should always be true